### DISCUSSION TITLE
Is it possible to get environment variable in the API?

### DISCUSSION THREAD

**Discussion by hdadr** [vercel/next.js#29516] [Help]
in **pages/api/send** api method i want to use a secret key for signing. Is it possible to get env variables from the .env.local file to this api without leaking it to the client side?
If so, how?

**Comment by maxtotheguenther** [comment]
@hdadr 
Thats totally possible.
Documentation:
https://nextjs.org/docs/basic-features/environment-variables
Just add following entry to your .env.local
SECRET=secret
This env variable will not be exposed to the client but is accessible on the server via process.env.SECRET
To expose this variable to the client you have to add the prefix NEXT_PUBLIC_
